elements/lisk-chain/test/unit/block.spec.ts

```diff
@@ -228,4 +228,186 @@ describe('block', () => {
 			});
 		});
 	});
+
+	describe('validateGenesis', () => {
+		let block: Block;
+		let tx: Transaction;
+		let assetList: BlockAsset[];
+		let blockAssets: BlockAssets;
+
+		beforeEach(() => {
+			assetList = [
+				{
+					moduleID: 3,
+					data: getRandomBytes(64),
+				},
+				{
+					moduleID: 6,
+					data: getRandomBytes(64),
+				},
+			];
+			blockAssets = new BlockAssets(assetList);
+			tx = getTransaction();
+		});
+
+		describe('when all values are valid', () => {
+			it('should not throw error', async () => {
+				// Arrange
+				block = await createValidDefaultBlock({ payload: [], assets: blockAssets });
+				// Act & assert
+				expect(() => block.validateGenesis()).not.toThrow();
+			});
+		});
+
+		describe('when payload is not empty', () => {
+			it('should throw error', async () => {
+				// Arrange
+				const txs = new Array(20).fill(0).map(() => tx);
+				block = await createValidDefaultBlock({ payload: txs, assets: blockAssets });
+				// Act & assert
+				expect(() => block.validateGenesis()).toThrow('Payload length must be zero');
+			});
+		});
+
+		describe('when an asset data has size more than the limit', () => {
+			it(`should not throw error when asset data length is greater than ${MAX_ASSET_DATA_SIZE_BYTES}`, async () => {
+				// Arrange
+				const assets = [
+					{
+						moduleID: 3,
+						data: getRandomBytes(64),
+					},
+					{
+						moduleID: 4,
+						data: getRandomBytes(128),
+					},
+				];
+				block = await createValidDefaultBlock({ assets: new BlockAssets(assets) });
+				// Act & assert
+				expect(() => block.validateGenesis()).not.toThrow();
+			});
+
+			it(`should pass when asset data length is equal or less than ${MAX_ASSET_DATA_SIZE_BYTES}`, async () => {
+				// Arrange
+				const assets = [
+					{
+						moduleID: 3,
+						data: getRandomBytes(64),
+					},
+					{
+						moduleID: 4,
+						data: getRandomBytes(64),
+					},
+				];
+				block = await createValidDefaultBlock({ assets: new BlockAssets(assets) });
+				// Act & assert
+				expect(block.validateGenesis()).toBeUndefined();
+			});
+		});
+
+		describe('when the assets are not sorted by moduleID', () => {
```

Comment:
i think block assets logic are moved to BlockAssets, so the test should also moved/added to the block assets spec

Reply:
yea but since it was not part of the original estimation would be better to move in a separate issue. Also, there are multiple places where the unit tests were added in the parent function instead of the owner(`./validate`). Also need to remove the old tests & usages for the same.

Reply:
make sense, I'll move the tests as per your suggestion.